oauthproxy.go

```diff
@@ -943,17 +953,20 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 		return nil, ErrNeedsLogin
 	}
 
-	invalidEmail := session != nil && session.Email != "" && !p.Validator(session.Email)
-	invalidGroups := session != nil && !p.validateGroups(session.Groups)
+	invalidEmail := session.Email != "" && !p.Validator(session.Email)
+	authorized, err := p.provider.Authorize(req.Context(), session)
+	if err != nil {
+		logger.Errorf("Error with authorization: %v", err)
+	}
 
-	if invalidEmail || invalidGroups {
-		logger.Printf(session.Email, req, logger.AuthFailure, "Invalid authentication via session: removing session %s", session)
+	if invalidEmail || !authorized {
+		logger.PrintAuthf(session.Email, req, logger.AuthFailure, "Invalid authorization via session: removing session %s", session)
 		// Invalid session, clear it
 		err := p.ClearSessionCookie(rw, req)
```

Comment:
Maybe a note for another time, but I wonder why we log the user out here, might be better to give them the option to log out rather than force logging them out, they could end up in a redirect loop if they do 🤔

Reply:
Yeah - I took note of the other PRs mentioning the infinite redirect issue. I think this change asserting authorization at the end of the `Redeem` stage covers that now:

https://github.com/oauth2-proxy/oauth2-proxy/pull/797/files#diff-7e1e132991c10dc0110f813e8e1057f24288bfb9d7d5bdd54d468c496e01110bR863

Reply:
Oh... are you talking infinite loop cause while they are logged out here they likely still have an IdP session/cookie that is still valid that will blow past the sign-in stage?

Reply:
Yeah the latter, we log out, redirect to IDP, pass back to callback, clear session, redirect to idp. I wonder if we need to actually show a 403 screen in this case? Do we do that already?

Reply:
I think it completely depends on if the user has the sign-in page enabled (so they wouldn't get the infinite redirect cause they have to click the button).

Or if they disabled it (potential infinite redirect loop).